iwyu_output.cc

```diff
@@ -1400,6 +1403,20 @@ void ProcessFullUse(OneUse* use,
       return;
     }
   }
+
+  // (B8) Discard uses of NamespaceDecls where the file also uses
+  // symbols within that namespace. IWYU for the symbol will also
+  // satisfy IWYU for the containing namespace(s).
+  if (const auto* ns_decl = dyn_cast<NamespaceDecl>(use->decl())) {
+    if (std::any_of(all_uses.begin(), all_uses.end(),
+                    [ns_decl](const OneUse& use) {
+                      const NamedDecl* decl = use.decl();
+                      return decl && IsInNamespace(decl, ns_decl);
```

Comment:
Feels like we should also make sure `decl != ns_decl` here...? I think it currently falls through a bit by accident, because `IsInNamespace` will start searching from the nearest declcontext of the passed-in decl, so it misses itself. Or maybe that's intentional and just subtle?

I'm thinking that it might be nice to make it more explicit.

Reply:
I think that's intentional - I'd expect `IsInNamespace(ns_decl, ns_decl)` to return false. Happy to put in the explicit check though.

Reply:
I think it might even make sense as an optimization to `IsInNamespace`, but never mind, I saw you added an explicit check. I feel safe now.